prometheus/promhttp/http.go

```diff
@@ -371,6 +376,21 @@ type HandlerOpts struct {
 	// (which changes the identity of the resulting series on the Prometheus
 	// server).
 	EnableOpenMetrics bool
+	// If 'EnableOpenMetrics' is true, 'EnableOpenMetricsCreatedMetrics' allows
+	// to add extra '_created' lines for counters, histograms and summaries,
+	// as defined in the OpenMetrics specification (see
+	// https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#counter-1).
+	// Created timestamps are used to improve the accuracy of reset detection,
+	// but be aware that it also increases the size of the payload.
+	//
+	// Prometheus introduced the feature flag 'created-timestamp-zero-ingestion'
+	// in version 2.50.0, but with support limited to the Prometheus protobuf
+	// format. Starting in Prometheus XXXX, the feature flag will be extended
+	// to the OpenMetrics text format. If using Prometheus XXXX or later, it
+	// is recommended to enable the feature flag in Prometheus, otherwise enabling
+	// _created lines will result in increased cardinality and no improvements
+	// in reset detection.
+	EnableOpenMetricsCreatedMetrics bool
```

Comment:
Nice. I wonder if we want to keep this or perhaps deprecate "EnableOpenMetrics" and have "OpenMetrics: *OpenMetricOpts" to add more stuff. We might need this to control:

* Created metrics in text
* Version
* ?

just an idea to not have trillion bool flags.

Reply:
Good idea 👍